features/step_definitions/machine.rb

```diff
@@ -71,6 +71,9 @@
     end
     
     machines_waiting_delete.each do | machine |
+      step %Q/I successfully merge patch resource "machine\/#{machine.name}" with:/, table(%{
+         | {"metadata":{"finalizers":[]}} |
+      })
       machine.ensure_deleted(user: user, wait: 1200)
       raise "Unable to delete machine #{machine.name}" if machine.exists?(user: user)
```

Comment:
Do we need this step? Why machine would exist in case `ensure_deleted` was successful? Some bug in `ensure_deleted`?

Can you post a log from error with your new code?

Reply:
@akostadinov It's bug in Machine Deletion. In some cases, machine cannot be deleted, stuck in Deleting status no matter how long we wait so that will cause finish prematurely. If we remove finalizers manually, the machine could be deleted.
log: http://pastebin.test.redhat.com/883804

Reply:
I don't ask about removing finalizers. I'm asking why do we need an additional check after `ensure_deleted` to check machine exists? I think `ensure_deleted` makes sure that's already the case.

Reply:
Thank you, got it. Yes, we don't need this step, will remove.  This is the error log if machine couldn't be deleted.
```
      [03:32:52] ERROR> Test Execution will finish prematurely.
      BushSlicer::Machine machineset-clone-28108-6rt5b did not disappear within 1200 sec (RuntimeError)
      /home/jenkins/workspace/Runner-v3/lib/openshift/resource.rb:184:in `ensure_deleted'
      /home/jenkins/workspace/Runner-v3/features/step_definitions/machine.rb:74:in `block (3 levels) in <top (required)>'
      /home/jenkins/workspace/Runner-v3/features/step_definitions/machine.rb:73:in `each'
      /home/jenkins/workspace/Runner-v3/features/step_definitions/machine.rb:73:in `block (2 levels) in <top (required)>'
      /home/jenkins/workspace/Runner-v3/lib/world.rb:432:in `block in after_scenario'
      /home/jenkins/workspace/Runner-v3/lib/world.rb:432:in `reverse_each'
      /home/jenkins/workspace/Runner-v3/lib/world.rb:432:in `after_scenario'
      /home/jenkins/workspace/Runner-v3/lib/manager.rb:35:in `clean_up'
      /home/jenkins/workspace/Runner-v3/features/support/env.rb:67:in `block (2 levels) in <top (required)>'
      /home/jenkins/workspace/Runner-v3/lib/base_helper.rb:131:in `capture_error'
      /home/jenkins/workspace/Runner-v3/features/support/env.rb:65:in `After'
```

Reply:
I think we can remove the line after ensure_deleted.

Reply:
Cool, please remove this line and move the test scenario somewhere under `features/test` directory with a proper comment that it is testing machine removal when finalizers are used. Then we can merge.

Thank you.

Reply:
@akostadinov Have updated. Then how to deal with the new code error: 
```
[09:07:51] ERROR> Test Execution will finish prematurely.
      patch failed to apply at []! machineset.machine.openshift.io "machineset-invalid" deleted
       (RuntimeError)
      /home/sunny/code/verification-tests/features/step_definitions/cli.rb:243:in `/^(?:(as admin) )?I successfully(?: (\w+|<%=.+?%>))? patch resource "(.*)\/(.*)" with:$/'
```
log: http://pastebin.test.redhat.com/883804

Reply:
It seems the patch is against the machineset, it should not.